File: apps/server/server/Models/Notification.cs
namespace ZhmApi.Models
{
    public enum NotificationType
    {
        AccessRequest,
        AccessRequestDecision,
        AppointmentReport,
        General
    }
    public class Notification
    {
        public int Id {get; set;}

        public int UserId {get; set;} 
        public string Message {get; set;} = null!;
        public bool IsRead { get; set;} = false;
        public NotificationType Type { get; set; }
        
        public int? AccessRequestId { get; set; }
        public AccessRequest? AccessRequest {get; set;} 
        public DateTime CreatedAt {get; set;} = DateTime.UtcNow;        
    }
}

File: apps/client/src/types/Notification.ts
export type NotificationType =
  | 'AccessRequest'
  | 'AccessRequestDecision'
  | 'AppointmentReport'
  | 'General'

export interface Notification {
  id: number
  message: string
  isRead: boolean
  type: NotificationType
  accessRequestId?: number
  createdAt: string
}
